### PR TITLE
Added acessibility to checkbox component

### PR DESCRIPTION
**Pull request by JanAckermann**
Add **id** and **for** properties to label
Add **aria-checked**, **aria-labelledby,**  **aria-disabled** property to input

### CODE REVIEW

**Comment by JammingBen** [comment]
Running `yarn test -u` should fix the tests

File: src/components/OcCheckbox.vue [hunk 1]
```diff
@@ -2,15 +2,17 @@
...
-      :class="$_ocCheckbox_classes"
+      :aria-labelledby="labelId"
+      :aria-checked="checked.toString()"
+      :aria-disabled="disabled.toString()"
+      :class="classes"
       :value="option"
       :disabled="disabled"
     />
-    <span v-if="!hideLabel" :aria-hidden="true" v-text="label" />
+    <span v-if="!hideLabel" :id="labelId" :aria-hidden="true" v-text="label" />
```
Comment (kulmann) [inline src/components/OcCheckbox.vue:15]: This is a bit more complicated. There are valid scenarios for both having a visual label and having no visual label. But in any scenario the checkbox needs to have an accessible label. I'd propose the following:
- get rid of the `hideLabel` prop
- keep the `label` prop (optional)
- add an `accessibleLabel` prop (required)
=> then have two spans: the one for `label` can be aria-hidden, the one for the `accessibleLabel` needs to be invisible for the user (use `oc-invisible-sr` class), but is the one actually used for the `aria-labelledby` prop on the checkbox. There will be situations then where the provided values for `label` and `accessibleLabel` are the same. I'm fine with that to not get an overcomplicated logic here.

Comment (JanAckermann) [reply src/components/OcCheckbox.vue:15]: I don't like the idea:
then we have the top label and 2 spans inside, will come up with a better solution

Comment (JammingBen) [reply src/components/OcCheckbox.vue:15]: Maybe I am missing something, but why not just remove `:aria-hidden="true"` from the current implementation? Then you can hide the label or not, but regardless it is used to label the checkbox (and therefore required ofc).

Comment (JanAckermann) [reply src/components/OcCheckbox.vue]: have a look

File: src/components/OcCheckbox.vue [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
+      :class="classes"
       :value="option"
       :disabled="disabled"
     />
-    <span v-if="!hideLabel" :aria-hidden="true" v-text="label" />
-  </label>
+    <label
+      :id="labelId"
+      :for="id"
+      :class="{ 'oc-invisible-sr': hideLabel === true }"
```
Comment (kulmann) [inline src/components/OcCheckbox.vue:18]:
```suggestion
      :class="{ 'oc-invisible-sr': hideLabel }"
```

File: changelog/unreleased/checkbox-acessibility [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Enhancement: Add aria properties to checkbox
```
Comment (kulmann) [inline changelog/unreleased/checkbox-acessibility:1]: Please add the `enhancement-` prefix to the changelog filename.

File: src/components/OcCheckbox.vue [hunk 4]
```diff
@@ -1,17 +1,17 @@
 <template>
-  <label :for="id" :class="{ 'oc-cursor-pointer': !disabled }">
+  <span :class="{ 'oc-cursor-pointer': !disabled }">
```
Comment (kulmann) [inline src/components/OcCheckbox.vue:2]: As with the radiobutton, please move the `oc-cursor-pointer` class to the label and all the conditionally applied classes of the label into a computed property.

**Review by kulmann — CHANGES_REQUESTED** [review]

File: src/components/OcCheckbox.vue [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
-      :aria-label="label"
-      :class="$_ocCheckbox_classes"
+      :aria-checked="checked.toString()"
+      :class="classes"
       :value="option"
       :disabled="disabled"
     />
-    <span v-if="!hideLabel" :aria-hidden="true" v-text="label" />
-  </label>
+    <label :for="id" :class="{ 'oc-invisible-sr': hideLabel }" v-text="label" />
```
Comment (kulmann) [inline src/components/OcCheckbox.vue:13]: pointer class on the label got lost now.

**Review by kulmann — CHANGES_REQUESTED** [review]

**Comment by JanAckermann** [comment]
Fucked something up while rebasing :-1:

**Review by kulmann — APPROVED** [review]
eslint doesn't agree. ;-) but code LGTM 👍

**Pull request merged**